### ISSUE TITLE
Automatically generate changelog

### ISSUE THREAD

**Issue by pavolloffay** [jaegertracing/jaeger-client-java#629]
After #58 we can release by pushing a tag to the repository. The next step is automatically to generate changelog.  

I have played with https://github.com/github-changelog-generator/github-changelog-generator. It looks to work fine but the structure is different to what we have right now.

I am adding otput for this repository for the following execution
```
docker run -it --rm -u "$(id -u)" -v "${PWD}:/usr/local/src/your-app" ferrarimarco/github-changelog-generator:1.14.3 -u jaegertracing -p jaeger-client-java --token XXXXXXXXX --release-branch master --no-issues --no-compare-link --unreleased-only true  --exclude-labels "no-changelog,duplicate,question,invalid,wontfix" -o unreleased.md
```

* trivial PRs can be exluded by `--exclude-labels exclude`
* It shows PR titles not commits - it's good we can change that after merge
* the order is based on PR number not merge time - I don't like it but it's acceptable https://github.com/github-changelog-generator/github-changelog-generator/issues/719
* breaking changes are supported in latest `rc` version - they will be grouped in one section.

--------------------------------------------------------------------------------

# Change Log

## [Unreleased](https://github.com/jaegertracing/jaeger-client-java/tree/HEAD)

**Merged pull requests:**

- Simplify release process [\#627](https://github.com/jaegertracing/jaeger-client-java/pull/627) ([pavolloffay](https://github.com/pavolloffay))

## [v0.35.2](https://github.com/jaegertracing/jaeger-client-java/tree/v0.35.2) (2019-05-06)
**Merged pull requests:**

- Fix nexus-publish config [\#621](https://github.com/jaegertracing/jaeger-client-java/pull/621) ([marcphilipp](https://github.com/marcphilipp))
- Binary format implementation [\#613](https://github.com/jaegertracing/jaeger-client-java/pull/613) ([ColinSullivan1](https://github.com/ColinSullivan1))

## [v0.35.1](https://github.com/jaegertracing/jaeger-client-java/tree/v0.35.1) (2019-05-02)
**Merged pull requests:**

- Update java-tracerresolver to version 0.1.6 [\#620](https://github.com/jaegertracing/jaeger-client-java/pull/620) ([Fox32](https://github.com/Fox32))
- Use publish task in travis deploy goal [\#619](https://github.com/jaegertracing/jaeger-client-java/pull/619) ([pavolloffay](https://github.com/pavolloffay))
- Upgrade gradle to 5.0 and use new maven-publish [\#618](https://github.com/jaegertracing/jaeger-client-java/pull/618) ([pavolloffay](https://github.com/pavolloffay))
- Revert "Prepare release 0.35.0" [\#617](https://github.com/jaegertracing/jaeger-client-java/pull/617) ([pavolloffay](https://github.com/pavolloffay))

## [v0.35.0](https://github.com/jaegertracing/jaeger-client-java/tree/v0.35.0) (2019-04-29)
**Merged pull requests:**

- Prepare release 0.35.0 [\#616](https://github.com/jaegertracing/jaeger-client-java/pull/616) ([pavolloffay](https://github.com/pavolloffay))
- Preparing release 0.35.0 [\#615](https://github.com/jaegertracing/jaeger-client-java/pull/615) ([pavolloffay](https://github.com/pavolloffay))
- Concurrency improvements to RemoteControlledSampler [\#609](https://github.com/jaegertracing/jaeger-client-java/pull/609) ([yborovikov](https://github.com/yborovikov))
- Back to development [\#603](https://github.com/jaegertracing/jaeger-client-java/pull/603) ([jpkrohling](https://github.com/jpkrohling))
- Adding support for opentracing 0.32 [\#567](https://github.com/jaegertracing/jaeger-client-java/pull/567) ([dougEfresh](https://github.com/dougEfresh))

## [v0.34.0](https://github.com/jaegertracing/jaeger-client-java/tree/v0.34.0) (2019-03-08)
**Merged pull requests:**

- Prepare release 0.34.0 [\#597](https://github.com/jaegertracing/jaeger-client-java/pull/597) ([jpkrohling](https://github.com/jpkrohling))
- Exclude transitive Apache httpclient dependencies from libthrift [\#596](https://github.com/jaegertracing/jaeger-client-java/pull/596) ([t-8ch](https://github.com/t-8ch))
- Handle all exceptions during queue processing [\#591](https://github.com/jaegertracing/jaeger-client-java/pull/591) ([dignati](https://github.com/dignati))
- Updated thrift to 0.12 [\#587](https://github.com/jaegertracing/jaeger-client-java/pull/587) ([jpkrohling](https://github.com/jpkrohling))
- \#585 offer manual shutdown of jaeger tracer [\#586](https://github.com/jaegertracing/jaeger-client-java/pull/586) ([supersven](https://github.com/supersven))
- Back to development [\#579](https://github.com/jaegertracing/jaeger-client-java/pull/579) ([pavolloffay](https://github.com/pavolloffay))

## [v0.33.1](https://github.com/jaegertracing/jaeger-client-java/tree/v0.33.1) (2019-01-03)
**Merged pull requests:**

- Prepare release 0.33.1 [\#578](https://github.com/jaegertracing/jaeger-client-java/pull/578) ([pavolloffay](https://github.com/pavolloffay))
- Back to development [\#576](https://github.com/jaegertracing/jaeger-client-java/pull/576) ([pavolloffay](https://github.com/pavolloffay))

## [v0.33.0](https://github.com/jaegertracing/jaeger-client-java/tree/v0.33.0) (2018-12-14)
**Merged pull requests:**

- Prepare release 0.33.0 [\#575](https://github.com/jaegertracing/jaeger-client-java/pull/575) ([objectiser](https://github.com/objectiser))
- Point to tchannel-based docker-compose [\#573](https://github.com/jaegertracing/jaeger-client-java/pull/573) ([yurishkuro](https://github.com/yurishkuro))
- Update reporter success metric on append in RemoteReporter \(\#211\) [\#572](https://github.com/jaegertracing/jaeger-client-java/pull/572) ([ikurovsky](https://github.com/ikurovsky))
- Always close HTTP responses in HttpSender [\#571](https://github.com/jaegertracing/jaeger-client-java/pull/571) ([mpetazzoni](https://github.com/mpetazzoni))
- Normalise metric names [\#564](https://github.com/jaegertracing/jaeger-client-java/pull/564) ([objectiser](https://github.com/objectiser))
- Prepare for next development [\#563](https://github.com/jaegertracing/jaeger-client-java/pull/563) ([pavolloffay](https://github.com/pavolloffay))
- Refer to valid `sampler type` values \(doc link\) [\#557](https://github.com/jaegertracing/jaeger-client-java/pull/557) ([sjoerdtalsma](https://github.com/sjoerdtalsma))

## [v0.32.0](https://github.com/jaegertracing/jaeger-client-java/tree/v0.32.0) (2018-10-15)
**Merged pull requests:**

- Remove nexus staging profile id [\#561](https://github.com/jaegertracing/jaeger-client-java/pull/561) ([pavolloffay](https://github.com/pavolloffay))
- Fix profile not being string [\#560](https://github.com/jaegertracing/jaeger-client-java/pull/560) ([pavolloffay](https://github.com/pavolloffay))
- Prepare release 0.32.0 [\#559](https://github.com/jaegertracing/jaeger-client-java/pull/559) ([pavolloffay](https://github.com/pavolloffay))
- Define nexus staging id explicitly [\#558](https://github.com/jaegertracing/jaeger-client-java/pull/558) ([pavolloffay](https://github.com/pavolloffay))
- Fix repeated resolving of sender in configuration [\#555](https://github.com/jaegertracing/jaeger-client-java/pull/555) ([pavolloffay](https://github.com/pavolloffay))
- Allow parsing configuration codec string programatically [\#550](https://github.com/jaegertracing/jaeger-client-java/pull/550) ([pavolloffay](https://github.com/pavolloffay))
- Close tracer on JVM shutdown [\#546](https://github.com/jaegertracing/jaeger-client-java/pull/546) ([jpkrohling](https://github.com/jpkrohling))
- Fix sampling when `jaeger-baggage` header is given [\#542](https://github.com/jaegertracing/jaeger-client-java/pull/542) ([yurishkuro](https://github.com/yurishkuro))
- Back to development 0.31.1-SNAPSHOT [\#540](https://github.com/jaegertracing/jaeger-client-java/pull/540) ([jpkrohling](https://github.com/jpkrohling))
- Implement TracerFactory [\#527](https://github.com/jaegertracing/jaeger-client-java/pull/527) ([jpkrohling](https://github.com/jpkrohling))
- Support for 128 bits trace ids [\#507](https://github.com/jaegertracing/jaeger-client-java/pull/507) ([hypnoce](https://github.com/hypnoce))

## [v0.31.0](https://github.com/jaegertracing/jaeger-client-java/tree/v0.31.0) (2018-08-28)
**Merged pull requests:**

- Prepare release 0.31.0 [\#538](https://github.com/jaegertracing/jaeger-client-java/pull/538) ([yurishkuro](https://github.com/yurishkuro))
- Back to development 0.30.7-SNAPSHOT [\#535](https://github.com/jaegertracing/jaeger-client-java/pull/535) ([jpkrohling](https://github.com/jpkrohling))
- Support `jaeger-baggage` header for ad-hoc baggage [\#525](https://github.com/jaegertracing/jaeger-client-java/pull/525) ([yurishkuro](https://github.com/yurishkuro))

## [v0.30.6](https://github.com/jaegertracing/jaeger-client-java/tree/v0.30.6) (2018-08-23)
**Merged pull requests:**

- Prepare release 0.30.6 [\#534](https://github.com/jaegertracing/jaeger-client-java/pull/534) ([jpkrohling](https://github.com/jpkrohling))
- Add specification version to jars, and impl-title/version & spec vers… [\#533](https://github.com/jaegertracing/jaeger-client-java/pull/533) ([objectiser](https://github.com/objectiser))
- Fine-tune bootstrap logging [\#531](https://github.com/jaegertracing/jaeger-client-java/pull/531) ([jpkrohling](https://github.com/jpkrohling))
- Back to development 0.30.6-SNAPSHOT [\#530](https://github.com/jaegertracing/jaeger-client-java/pull/530) ([jpkrohling](https://github.com/jpkrohling))

## [v0.30.5](https://github.com/jaegertracing/jaeger-client-java/tree/v0.30.5) (2018-08-22)
**Merged pull requests:**

- Prepare release 0.30.5 [\#529](https://github.com/jaegertracing/jaeger-client-java/pull/529) ([jpkrohling](https://github.com/jpkrohling))
- Mark gson field in HttpSamplingManager as ignored for toString [\#528](https://github.com/jaegertracing/jaeger-client-java/pull/528) ([jpkrohling](https://github.com/jpkrohling))
- Move jaeger-thrift shadow jar to its own classifier [\#526](https://github.com/jaegertracing/jaeger-client-java/pull/526) ([jpkrohling](https://github.com/jpkrohling))
- Use SenderFactory classloader for ServiceLoader [\#523](https://github.com/jaegertracing/jaeger-client-java/pull/523) ([dbuchhorn](https://github.com/dbuchhorn))
- Reduce visibility of TextMapCodec.contextFromString to package scope [\#519](https://github.com/jaegertracing/jaeger-client-java/pull/519) ([isaachier](https://github.com/isaachier))
- Move contextFromString out of JaegerSpanContext into TextMapCodec [\#517](https://github.com/jaegertracing/jaeger-client-java/pull/517) ([isaachier](https://github.com/isaachier))
- Back to development 0.30.5-SNAPSHOT [\#514](https://github.com/jaegertracing/jaeger-client-java/pull/514) ([jpkrohling](https://github.com/jpkrohling))
- Allow subclassing of common classes [\#509](https://github.com/jaegertracing/jaeger-client-java/pull/509) ([isaachier](https://github.com/isaachier))

## [v0.30.4](https://github.com/jaegertracing/jaeger-client-java/tree/v0.30.4) (2018-08-13)
**Merged pull requests:**

- Prepare release 0.30.4 [\#513](https://github.com/jaegertracing/jaeger-client-java/pull/513) ([jpkrohling](https://github.com/jpkrohling))
- Add possibility to specify serviceName with 'fromEnv' [\#510](https://github.com/jaegertracing/jaeger-client-java/pull/510) ([jpkrohling](https://github.com/jpkrohling))
- Back to development 0.30.4-SNAPSHOT [\#505](https://github.com/jaegertracing/jaeger-client-java/pull/505) ([jpkrohling](https://github.com/jpkrohling))

## [v0.30.3](https://github.com/jaegertracing/jaeger-client-java/tree/v0.30.3) (2018-07-26)
**Merged pull requests:**

- Prepare release 0.30.3 [\#504](https://github.com/jaegertracing/jaeger-client-java/pull/504) ([jpkrohling](https://github.com/jpkrohling))
- Fix concurrent modification on logs and tags [\#500](https://github.com/jaegertracing/jaeger-client-java/pull/500) ([pavolloffay](https://github.com/pavolloffay))
- Fixes \#494 - removed unwanted classes from jaeger-thrift shadowed jar [\#498](https://github.com/jaegertracing/jaeger-client-java/pull/498) ([mdvorak](https://github.com/mdvorak))
- Back to development 0.30.3-SNAPSHOT [\#492](https://github.com/jaegertracing/jaeger-client-java/pull/492) ([jpkrohling](https://github.com/jpkrohling))

## [v0.30.2](https://github.com/jaegertracing/jaeger-client-java/tree/v0.30.2) (2018-07-13)
**Merged pull requests:**

- Prepare release 0.30.2 [\#491](https://github.com/jaegertracing/jaeger-client-java/pull/491) ([jpkrohling](https://github.com/jpkrohling))
- Fix Zipkin readme [\#489](https://github.com/jaegertracing/jaeger-client-java/pull/489) ([jpkrohling](https://github.com/jpkrohling))
- Fix issue where warning message is generated each time a null referen… [\#487](https://github.com/jaegertracing/jaeger-client-java/pull/487) ([objectiser](https://github.com/objectiser))
- Fix jaeger-client dependency jaeger-thrift no-shadow artifact [\#486](https://github.com/jaegertracing/jaeger-client-java/pull/486) ([jpkrohling](https://github.com/jpkrohling))
- Back to development 0.30.2-SNAPSHOT [\#485](https://github.com/jaegertracing/jaeger-client-java/pull/485) ([jpkrohling](https://github.com/jpkrohling))

## [v0.30.1](https://github.com/jaegertracing/jaeger-client-java/tree/v0.30.1) (2018-07-10)
**Merged pull requests:**

- Specify custom sender for tests in the module jaeger-client [\#483](https://github.com/jaegertracing/jaeger-client-java/pull/483) ([jpkrohling](https://github.com/jpkrohling))
- Prepare release 0.30.1 [\#482](https://github.com/jaegertracing/jaeger-client-java/pull/482) ([jpkrohling](https://github.com/jpkrohling))
- Add "serialVersionUID" for exceptions to avoid InvalidClassException [\#481](https://github.com/jaegertracing/jaeger-client-java/pull/481) ([quaff](https://github.com/quaff))
- Fix wrong package declaration [\#480](https://github.com/jaegertracing/jaeger-client-java/pull/480) ([quaff](https://github.com/quaff))
- Add jaeger-client module [\#479](https://github.com/jaegertracing/jaeger-client-java/pull/479) ([jpkrohling](https://github.com/jpkrohling))
- Fix version retrieval from jaeger.properties [\#478](https://github.com/jaegertracing/jaeger-client-java/pull/478) ([jpkrohling](https://github.com/jpkrohling))
- Fix links to PRs on changelog for v0.30.0 [\#476](https://github.com/jaegertracing/jaeger-client-java/pull/476) ([jpkrohling](https://github.com/jpkrohling))
- Back to development 0.30.1-SNAPSHOT [\#475](https://github.com/jaegertracing/jaeger-client-java/pull/475) ([jpkrohling](https://github.com/jpkrohling))

## [v0.30.0](https://github.com/jaegertracing/jaeger-client-java/tree/v0.30.0) (2018-07-04)
**Merged pull requests:**

- Prepare release 0.30.0 [\#474](https://github.com/jaegertracing/jaeger-client-java/pull/474) ([jpkrohling](https://github.com/jpkrohling))
- Define some classes internal [\#470](https://github.com/jaegertracing/jaeger-client-java/pull/470) ([jpkrohling](https://github.com/jpkrohling))
- Change return types from Jaeger Span/Tracer/Context to Jaeger types [\#469](https://github.com/jaegertracing/jaeger-client-java/pull/469) ([jpkrohling](https://github.com/jpkrohling))
- Treat gauge.update\(\) parameter as new value, not delta [\#463](https://github.com/jaegertracing/jaeger-client-java/pull/463) ([mdouaihy](https://github.com/mdouaihy))
- Add javadoc badge [\#462](https://github.com/jaegertracing/jaeger-client-java/pull/462) ([jpkrohling](https://github.com/jpkrohling))
- Make jaeger-thrift's shaded JAR the default one [\#461](https://github.com/jaegertracing/jaeger-client-java/pull/461) ([jpkrohling](https://github.com/jpkrohling))
- Adjust thrift shadow configuration and version [\#458](https://github.com/jaegertracing/jaeger-client-java/pull/458) ([jpkrohling](https://github.com/jpkrohling))
- Make GuaranteedThroughputSampler public [\#457](https://github.com/jaegertracing/jaeger-client-java/pull/457) ([quaff](https://github.com/quaff))
- Upgrades internals to Zipkin v2 library [\#456](https://github.com/jaegertracing/jaeger-client-java/pull/456) ([adriancole](https://github.com/adriancole))
- Address build warnings [\#455](https://github.com/jaegertracing/jaeger-client-java/pull/455) ([jpkrohling](https://github.com/jpkrohling))
- Adding new issue and pull request template [\#451](https://github.com/jaegertracing/jaeger-client-java/pull/451) ([PikBot](https://github.com/PikBot))
- Remove dependency from jaeger-core to jaeger-thrift [\#449](https://github.com/jaegertracing/jaeger-client-java/pull/449) ([jpkrohling](https://github.com/jpkrohling))
- Back to development 0.29.1-SNAPSHOT [\#447](https://github.com/jaegertracing/jaeger-client-java/pull/447) ([pavolloffay](https://github.com/pavolloffay))
- Revert "Fix travis release" [\#446](https://github.com/jaegertracing/jaeger-client-java/pull/446) ([pavolloffay](https://github.com/pavolloffay))
- Fix travis release [\#443](https://github.com/jaegertracing/jaeger-client-java/pull/443) ([pavolloffay](https://github.com/pavolloffay))

## [v0.29.0](https://github.com/jaegertracing/jaeger-client-java/tree/v0.29.0) (2018-06-07)
**Merged pull requests:**

- Rename 'io.jaegertracing.Span' to 'io.jaegertracing.JaegerSpan' [\#454](https://github.com/jaegertracing/jaeger-client-java/pull/454) ([jpkrohling](https://github.com/jpkrohling))
- Preparing release 0.29.0 [\#442](https://github.com/jaegertracing/jaeger-client-java/pull/442) ([pavolloffay](https://github.com/pavolloffay))
- Add endpoint javadoc for HTTP and udp senders [\#441](https://github.com/jaegertracing/jaeger-client-java/pull/441) ([pavolloffay](https://github.com/pavolloffay))
- Deprecate B3 codec contructor [\#440](https://github.com/jaegertracing/jaeger-client-java/pull/440) ([pavolloffay](https://github.com/pavolloffay))
- Add baggage to B3 codec [\#438](https://github.com/jaegertracing/jaeger-client-java/pull/438) ([pavolloffay](https://github.com/pavolloffay))
- Load metrics factory service from classpath if available [\#433](https://github.com/jaegertracing/jaeger-client-java/pull/433) ([jpkrohling](https://github.com/jpkrohling))
- Fix xdock image build on travis [\#432](https://github.com/jaegertracing/jaeger-client-java/pull/432) ([pavolloffay](https://github.com/pavolloffay))
- Remove deprecated API - HttpSender and RemoteBaggMana [\#431](https://github.com/jaegertracing/jaeger-client-java/pull/431) ([pavolloffay](https://github.com/pavolloffay))
- Back to development [\#430](https://github.com/jaegertracing/jaeger-client-java/pull/430) ([pavolloffay](https://github.com/pavolloffay))

## [v0.28.0](https://github.com/jaegertracing/jaeger-client-java/tree/v0.28.0) (2018-05-24)
**Merged pull requests:**

- Release 0.28.0 [\#429](https://github.com/jaegertracing/jaeger-client-java/pull/429) ([pavolloffay](https://github.com/pavolloffay))
- Remove unused exception [\#428](https://github.com/jaegertracing/jaeger-client-java/pull/428) ([pavolloffay](https://github.com/pavolloffay))
- Change changelog.rst to markdown [\#427](https://github.com/jaegertracing/jaeger-client-java/pull/427) ([pavolloffay](https://github.com/pavolloffay))
- Use string builder instead of string format [\#425](https://github.com/jaegertracing/jaeger-client-java/pull/425) ([pavolloffay](https://github.com/pavolloffay))
- Add builder patter and deprecate some APIs [\#424](https://github.com/jaegertracing/jaeger-client-java/pull/424) ([pavolloffay](https://github.com/pavolloffay))
- Set executable flag on script publish-crossdock.sh [\#422](https://github.com/jaegertracing/jaeger-client-java/pull/422) ([pavolloffay](https://github.com/pavolloffay))
- fix deploy stage on travis [\#421](https://github.com/jaegertracing/jaeger-client-java/pull/421) ([pavolloffay](https://github.com/pavolloffay))
- Bump gradle wrapper to 4.7 [\#419](https://github.com/jaegertracing/jaeger-client-java/pull/419) ([pavolloffay](https://github.com/pavolloffay))
- Bump versions of gradle plugins [\#418](https://github.com/jaegertracing/jaeger-client-java/pull/418) ([pavolloffay](https://github.com/pavolloffay))
- Do not use deprecated Tracer.Builder constructor internally [\#417](https://github.com/jaegertracing/jaeger-client-java/pull/417) ([pavolloffay](https://github.com/pavolloffay))
- Do not use deprecated APIs internally [\#416](https://github.com/jaegertracing/jaeger-client-java/pull/416) ([pavolloffay](https://github.com/pavolloffay))
- Fix flaky test in RemoteReporterTest [\#415](https://github.com/jaegertracing/jaeger-client-java/pull/415) ([pavolloffay](https://github.com/pavolloffay))
- Remove deprecated APIs [\#414](https://github.com/jaegertracing/jaeger-client-java/pull/414) ([pavolloffay](https://github.com/pavolloffay))
- Do not use deprecated APIs internally [\#412](https://github.com/jaegertracing/jaeger-client-java/pull/412) ([pavolloffay](https://github.com/pavolloffay))
- Polish configuration API [\#411](https://github.com/jaegertracing/jaeger-client-java/pull/411) ([pavolloffay](https://github.com/pavolloffay))
- Remove dependency on brave-http [\#410](https://github.com/jaegertracing/jaeger-client-java/pull/410) ([pavolloffay](https://github.com/pavolloffay))
- Remove tchannel from crossdock [\#405](https://github.com/jaegertracing/jaeger-client-java/pull/405) ([yurishkuro](https://github.com/yurishkuro))
- Add Node.js client to crossdock matrix [\#404](https://github.com/jaegertracing/jaeger-client-java/pull/404) ([yurishkuro](https://github.com/yurishkuro))
- Remove tchannel as tested transport in crossdock [\#403](https://github.com/jaegertracing/jaeger-client-java/pull/403) ([yurishkuro](https://github.com/yurishkuro))
- Define okhttp version in top level build.gradle, along with other dep… [\#402](https://github.com/jaegertracing/jaeger-client-java/pull/402) ([objectiser](https://github.com/objectiser))
- Set development version to 0.27.1-SNAPSHOT [\#401](https://github.com/jaegertracing/jaeger-client-java/pull/401) ([jpkrohling](https://github.com/jpkrohling))
- Implement Zipkin 2 JSON Sender [\#399](https://github.com/jaegertracing/jaeger-client-java/pull/399) ([keitwb](https://github.com/keitwb))
- Describe release process [\#392](https://github.com/jaegertracing/jaeger-client-java/pull/392) ([pavolloffay](https://github.com/pavolloffay))
- Use travis stages to simplify config [\#389](https://github.com/jaegertracing/jaeger-client-java/pull/389) ([pavolloffay](https://github.com/pavolloffay))

## [v0.27.0](https://github.com/jaegertracing/jaeger-client-java/tree/v0.27.0) (2018-04-18)
**Merged pull requests:**

- Preparing release 0.27.0 [\#400](https://github.com/jaegertracing/jaeger-client-java/pull/400) ([jpkrohling](https://github.com/jpkrohling))
- Enable the signing of the archives to be marked as not required if -P… [\#398](https://github.com/jaegertracing/jaeger-client-java/pull/398) ([objectiser](https://github.com/objectiser))
- Enable development and dependency versions to be overridden from the … [\#397](https://github.com/jaegertracing/jaeger-client-java/pull/397) ([objectiser](https://github.com/objectiser))
- Back to development [\#393](https://github.com/jaegertracing/jaeger-client-java/pull/393) ([pavolloffay](https://github.com/pavolloffay))
- Add tests for metrics exposed via prometheus [\#386](https://github.com/jaegertracing/jaeger-client-java/pull/386) ([kevinearls](https://github.com/kevinearls))

## [v0.27.0-RC1](https://github.com/jaegertracing/jaeger-client-java/tree/v0.27.0-RC1) (2018-04-11)
**Merged pull requests:**

- Pin release version 0.27.0-RC1 [\#391](https://github.com/jaegertracing/jaeger-client-java/pull/391) ([pavolloffay](https://github.com/pavolloffay))
- Re-encrypt sonatype user and password [\#388](https://github.com/jaegertracing/jaeger-client-java/pull/388) ([pavolloffay](https://github.com/pavolloffay))
- Add changelong entry for 0.27.0 [\#387](https://github.com/jaegertracing/jaeger-client-java/pull/387) ([pavolloffay](https://github.com/pavolloffay))
- Remove reference to JAEGER\_DISABLE\_GLOBAL\_TRACER [\#379](https://github.com/jaegertracing/jaeger-client-java/pull/379) ([Sharpie](https://github.com/Sharpie))
- \[jaeger/apache-http-instrumentation\] TracingRequestInterceptor gets span from http request context vs tracer scope manager [\#378](https://github.com/jaegertracing/jaeger-client-java/pull/378) ([dray92](https://github.com/dray92))
- Point IDL submodule git link to jaegertracing org [\#377](https://github.com/jaegertracing/jaeger-client-java/pull/377) ([pavolloffay](https://github.com/pavolloffay))
- HttpSenderBuilder define some methods as public [\#376](https://github.com/jaegertracing/jaeger-client-java/pull/376) ([pavolloffay](https://github.com/pavolloffay))
- Rename to io.jaegertracing [\#375](https://github.com/jaegertracing/jaeger-client-java/pull/375) ([pavolloffay](https://github.com/pavolloffay))
- Avoid direct access to apache thrift from jaeger-core via transitive … [\#374](https://github.com/jaegertracing/jaeger-client-java/pull/374) ([objectiser](https://github.com/objectiser))
- Set development version to 0.26.1-SNAPSHOT [\#373](https://github.com/jaegertracing/jaeger-client-java/pull/373) ([jpkrohling](https://github.com/jpkrohling))
- Preparing release 0.26.0 [\#372](https://github.com/jaegertracing/jaeger-client-java/pull/372) ([jpkrohling](https://github.com/jpkrohling))

## [v0.26.0](https://github.com/jaegertracing/jaeger-client-java/tree/v0.26.0) (2018-03-19)
**Merged pull requests:**

- Avoid host lookups if trace tags have already been provided [\#371](https://github.com/jaegertracing/jaeger-client-java/pull/371) ([huntc](https://github.com/huntc))
- Updating the ReporterConfig to set Sender Correctly [\#370](https://github.com/jaegertracing/jaeger-client-java/pull/370) ([natehart](https://github.com/natehart))
- Set development version to 0.25.1-SNAPSHOT [\#368](https://github.com/jaegertracing/jaeger-client-java/pull/368) ([jpkrohling](https://github.com/jpkrohling))
- Fix for issue 366, fix deprecated code in example [\#367](https://github.com/jaegertracing/jaeger-client-java/pull/367) ([kevinearls](https://github.com/kevinearls))

## [v0.25.0](https://github.com/jaegertracing/jaeger-client-java/tree/v0.25.0) (2018-03-07)
**Merged pull requests:**

- Preparing release 0.25.0 [\#365](https://github.com/jaegertracing/jaeger-client-java/pull/365) ([jpkrohling](https://github.com/jpkrohling))
- codecs resiliency, another approach [\#362](https://github.com/jaegertracing/jaeger-client-java/pull/362) ([tmszdmsk](https://github.com/tmszdmsk))
- b3 codec test readability improvement. [\#358](https://github.com/jaegertracing/jaeger-client-java/pull/358) ([tmszdmsk](https://github.com/tmszdmsk))
- Closes \#350 - Documented Micrometer integration [\#356](https://github.com/jaegertracing/jaeger-client-java/pull/356) ([jpkrohling](https://github.com/jpkrohling))
- Closes \#354 - Ignores B3 headers if invalid values are provided [\#355](https://github.com/jaegertracing/jaeger-client-java/pull/355) ([jpkrohling](https://github.com/jpkrohling))
- JerseyServerFilter\#JerseyServerFilter\(Tracer, TraceContext\) deprecated [\#352](https://github.com/jaegertracing/jaeger-client-java/pull/352) ([dray92](https://github.com/dray92))
- Closes \#338 - Deprecated StatsReporter [\#349](https://github.com/jaegertracing/jaeger-client-java/pull/349) ([jpkrohling](https://github.com/jpkrohling))
- Update micrometer version to 1.0.0 [\#348](https://github.com/jaegertracing/jaeger-client-java/pull/348) ([objectiser](https://github.com/objectiser))
- Update tracerresolver version to 0.1.4 to align with OT Java 0.31 [\#347](https://github.com/jaegertracing/jaeger-client-java/pull/347) ([objectiser](https://github.com/objectiser))
- Refactor public builder API [\#346](https://github.com/jaegertracing/jaeger-client-java/pull/346) ([pavolloffay](https://github.com/pavolloffay))
- Use SortedMap directly istead of sorting just keys [\#343](https://github.com/jaegertracing/jaeger-client-java/pull/343) ([pavolloffay](https://github.com/pavolloffay))
- Closes \#340 - Disables flaky test \(RemoteReporterTest\#testCloseWhenQueueFull\) [\#341](https://github.com/jaegertracing/jaeger-client-java/pull/341) ([jpkrohling](https://github.com/jpkrohling))
-  Closes \#298 - Metrics are now consistent with Go  [\#335](https://github.com/jaegertracing/jaeger-client-java/pull/335) ([jpkrohling](https://github.com/jpkrohling))
- Back to development [\#331](https://github.com/jaegertracing/jaeger-client-java/pull/331) ([dray92](https://github.com/dray92))
- Expand exception logs [\#168](https://github.com/jaegertracing/jaeger-client-java/pull/168) ([pavolloffay](https://github.com/pavolloffay))

## [v0.24.0](https://github.com/jaegertracing/jaeger-client-java/tree/v0.24.0) (2018-02-01)
**Merged pull requests:**

- update changelog [\#330](https://github.com/jaegertracing/jaeger-client-java/pull/330) ([dray92](https://github.com/dray92))
- Adding annotated filter priorities for Jersey filters [\#329](https://github.com/jaegertracing/jaeger-client-java/pull/329) ([dray92](https://github.com/dray92))
- Fix ClassCastException when a wrapped span impl is activated instead … [\#327](https://github.com/jaegertracing/jaeger-client-java/pull/327) ([objectiser](https://github.com/objectiser))
- Give Gradle 1gb of memory [\#324](https://github.com/jaegertracing/jaeger-client-java/pull/324) ([yurishkuro](https://github.com/yurishkuro))
- Enable info logs and stacktraces from gradle [\#323](https://github.com/jaegertracing/jaeger-client-java/pull/323) ([vprithvi](https://github.com/vprithvi))
- set travis wait time to 25 mins [\#322](https://github.com/jaegertracing/jaeger-client-java/pull/322) ([dray92](https://github.com/dray92))
- \[jaeger-client-java/apache instrumentation\] Completely remove apache instrumentation dependence on jaeger-context [\#320](https://github.com/jaegertracing/jaeger-client-java/pull/320) ([dray92](https://github.com/dray92))
- \[jaeger-client-java/jax-rs2\] Remove jaeger-context dependence for JAX-RS2 jaeger instrumentation [\#319](https://github.com/jaegertracing/jaeger-client-java/pull/319) ([dray92](https://github.com/dray92))
- Remove shadow dependency from jaeger-b3 [\#309](https://github.com/jaegertracing/jaeger-client-java/pull/309) ([vprithvi](https://github.com/vprithvi))
- This closes \#256 by better handling baggage null values [\#308](https://github.com/jaegertracing/jaeger-client-java/pull/308) ([bmozaffa](https://github.com/bmozaffa))

## [v0.23.0](https://github.com/jaegertracing/jaeger-client-java/tree/v0.23.0) (2018-01-13)
**Merged pull requests:**

- Upgrade to OpenTracing 0.31.0 [\#313](https://github.com/jaegertracing/jaeger-client-java/pull/313) ([yurishkuro](https://github.com/yurishkuro))
- Update Gradle to 4.4 [\#312](https://github.com/jaegertracing/jaeger-client-java/pull/312) ([vprithvi](https://github.com/vprithvi))
- Make apache context span key public [\#310](https://github.com/jaegertracing/jaeger-client-java/pull/310) ([dray92](https://github.com/dray92))
- Fix currentSpan\(\) when active span is null [\#307](https://github.com/jaegertracing/jaeger-client-java/pull/307) ([yurishkuro](https://github.com/yurishkuro))
- Clean up code style and javadoc warnings [\#305](https://github.com/jaegertracing/jaeger-client-java/pull/305) ([yurishkuro](https://github.com/yurishkuro))
- Remove concept of registering a GlobalTracer when getting a new tracer [\#302](https://github.com/jaegertracing/jaeger-client-java/pull/302) ([dray92](https://github.com/dray92))
- Fix incorrect development version prior to next release [\#301](https://github.com/jaegertracing/jaeger-client-java/pull/301) ([objectiser](https://github.com/objectiser))
- \[jaeger-client-java\] Span class does not report multiple time on multiple invocations of finish\(\) [\#299](https://github.com/jaegertracing/jaeger-client-java/pull/299) ([dray92](https://github.com/dray92))
- \[apache\] \[jax-rs2\] Instrumentation specific multiple span reporting Issue [\#297](https://github.com/jaegertracing/jaeger-client-java/pull/297) ([dray92](https://github.com/dray92))
- Configuration enabled support for B3 trace context propagation in add… [\#296](https://github.com/jaegertracing/jaeger-client-java/pull/296) ([objectiser](https://github.com/objectiser))
- Improve tests [\#292](https://github.com/jaegertracing/jaeger-client-java/pull/292) ([alshopov](https://github.com/alshopov))
- Iterate on Maps via Entry rather than key + retrieval of value [\#290](https://github.com/jaegertracing/jaeger-client-java/pull/290) ([alshopov](https://github.com/alshopov))
- Updates of runtime components [\#286](https://github.com/jaegertracing/jaeger-client-java/pull/286) ([alshopov](https://github.com/alshopov))
- Added auth options to HttpSender [\#285](https://github.com/jaegertracing/jaeger-client-java/pull/285) ([jpkrohling](https://github.com/jpkrohling))
- Skip upload to docker hub from forked repo PRs [\#284](https://github.com/jaegertracing/jaeger-client-java/pull/284) ([black-adder](https://github.com/black-adder))
- Small refactors [\#283](https://github.com/jaegertracing/jaeger-client-java/pull/283) ([alshopov](https://github.com/alshopov))
- Gradual updates to dependencies - no runtime changes [\#282](https://github.com/jaegertracing/jaeger-client-java/pull/282) ([alshopov](https://github.com/alshopov))
- Plugin for updates and update httpcomponents [\#281](https://github.com/jaegertracing/jaeger-client-java/pull/281) ([alshopov](https://github.com/alshopov))
- Use shared jaeger-compose in xdock [\#276](https://github.com/jaegertracing/jaeger-client-java/pull/276) ([pavolloffay](https://github.com/pavolloffay))

## [v0.22.0-RC2](https://github.com/jaegertracing/jaeger-client-java/tree/v0.22.0-RC2) (2017-10-26)
**Merged pull requests:**

- Update to Gradle 4.2.1 [\#274](https://github.com/jaegertracing/jaeger-client-java/pull/274) ([vprithvi](https://github.com/vprithvi))
- Splitting out the jaxrs2 filters [\#273](https://github.com/jaegertracing/jaeger-client-java/pull/273) ([dray92](https://github.com/dray92))
- Update errorprone and lombok and fix warnings and errors [\#270](https://github.com/jaegertracing/jaeger-client-java/pull/270) ([vprithvi](https://github.com/vprithvi))
- Add a ActiveSpanSource backed TraceContext [\#266](https://github.com/jaegertracing/jaeger-client-java/pull/266) ([vprithvi](https://github.com/vprithvi))
- Clean-up comments and code style references [\#264](https://github.com/jaegertracing/jaeger-client-java/pull/264) ([yurishkuro](https://github.com/yurishkuro))
- Update to Apache License v. 2 [\#262](https://github.com/jaegertracing/jaeger-client-java/pull/262) ([yurishkuro](https://github.com/yurishkuro))
- Make gradle use travis/docker-thrift/thrift script [\#261](https://github.com/jaegertracing/jaeger-client-java/pull/261) ([yurishkuro](https://github.com/yurishkuro))
- Use io.opentracing.GlobalTracer instead of singleton in Configuration [\#255](https://github.com/jaegertracing/jaeger-client-java/pull/255) ([vprithvi](https://github.com/vprithvi))
- Changelog for 0.22.0-RC1 [\#252](https://github.com/jaegertracing/jaeger-client-java/pull/252) ([pavolloffay](https://github.com/pavolloffay))
- Request for comments: Breaking apart functionality of TracingRequestInterceptor [\#248](https://github.com/jaegertracing/jaeger-client-java/pull/248) ([dray92](https://github.com/dray92))

## [v0.22.0-RC1](https://github.com/jaegertracing/jaeger-client-java/tree/v0.22.0-RC1) (2017-10-05)
**Merged pull requests:**

- Updated the signing env vars to match the new organization [\#249](https://github.com/jaegertracing/jaeger-client-java/pull/249) ([jpkrohling](https://github.com/jpkrohling))
- Disable javadoc job for jaeger-thrift [\#245](https://github.com/jaegertracing/jaeger-client-java/pull/245) ([vprithvi](https://github.com/vprithvi))
- Span builder should accept null in asChildOf [\#243](https://github.com/jaegertracing/jaeger-client-java/pull/243) ([pavolloffay](https://github.com/pavolloffay))
- Tracer implements Closeable [\#242](https://github.com/jaegertracing/jaeger-client-java/pull/242) ([pavolloffay](https://github.com/pavolloffay))
- Automatically close and release staging repository [\#241](https://github.com/jaegertracing/jaeger-client-java/pull/241) ([jpkrohling](https://github.com/jpkrohling))
- Series of small patches with small improvemts grouped by type [\#237](https://github.com/jaegertracing/jaeger-client-java/pull/237) ([alshopov](https://github.com/alshopov))
- Update changelog [\#234](https://github.com/jaegertracing/jaeger-client-java/pull/234) ([vprithvi](https://github.com/vprithvi))

## [v0.21.0](https://github.com/jaegertracing/jaeger-client-java/tree/v0.21.0) (2017-09-15)
**Merged pull requests:**

- Senders fix, do not use static thrift factory [\#233](https://github.com/jaegertracing/jaeger-client-java/pull/233) ([pavolloffay](https://github.com/pavolloffay))
- Add RemoteBaggageRestrictionManager [\#229](https://github.com/jaegertracing/jaeger-client-java/pull/229) ([black-adder](https://github.com/black-adder))
- Support span log fields in zipkin sender [\#226](https://github.com/jaegertracing/jaeger-client-java/pull/226) ([pvlugter](https://github.com/pvlugter))
- Add BaggageRestrictionManager [\#217](https://github.com/jaegertracing/jaeger-client-java/pull/217) ([black-adder](https://github.com/black-adder))

## [v0.20.6](https://github.com/jaegertracing/jaeger-client-java/tree/v0.20.6) (2017-07-28)
**Merged pull requests:**

- Change destination\(Object\) to destination\(File\) [\#224](https://github.com/jaegertracing/jaeger-client-java/pull/224) ([alshopov](https://github.com/alshopov))
- Use okhttp [\#223](https://github.com/jaegertracing/jaeger-client-java/pull/223) ([vprithvi](https://github.com/vprithvi))
- Refactor inner class to outer [\#221](https://github.com/jaegertracing/jaeger-client-java/pull/221) ([alshopov](https://github.com/alshopov))
- Small fix to docs [\#219](https://github.com/jaegertracing/jaeger-client-java/pull/219) ([alshopov](https://github.com/alshopov))
- Enable tracer level tags to be initialised from environment variable [\#215](https://github.com/jaegertracing/jaeger-client-java/pull/215) ([objectiser](https://github.com/objectiser))

## [v0.20.5](https://github.com/jaegertracing/jaeger-client-java/tree/v0.20.5) (2017-06-30)
**Merged pull requests:**

- Chop off the 'v' from the tag name [\#209](https://github.com/jaegertracing/jaeger-client-java/pull/209) ([jpkrohling](https://github.com/jpkrohling))

## [v0.20.4](https://github.com/jaegertracing/jaeger-client-java/tree/v0.20.4) (2017-06-30)
**Merged pull requests:**

- Upload artifacts for tags [\#208](https://github.com/jaegertracing/jaeger-client-java/pull/208) ([jpkrohling](https://github.com/jpkrohling))

## [v0.20.3](https://github.com/jaegertracing/jaeger-client-java/tree/v0.20.3) (2017-06-30)
**Merged pull requests:**

- Changed encrypted signing key [\#207](https://github.com/jaegertracing/jaeger-client-java/pull/207) ([jpkrohling](https://github.com/jpkrohling))

## [v0.20.2](https://github.com/jaegertracing/jaeger-client-java/tree/v0.20.2) (2017-06-29)
**Merged pull requests:**

- Fixed signing keys [\#206](https://github.com/jaegertracing/jaeger-client-java/pull/206) ([jpkrohling](https://github.com/jpkrohling))

## [v0.20.1](https://github.com/jaegertracing/jaeger-client-java/tree/v0.20.1) (2017-06-29)
**Merged pull requests:**

- Fix crossdock [\#203](https://github.com/jaegertracing/jaeger-client-java/pull/203) ([black-adder](https://github.com/black-adder))
- Automated the release process [\#201](https://github.com/jaegertracing/jaeger-client-java/pull/201) ([jpkrohling](https://github.com/jpkrohling))
- Bump gradle to 4.0 [\#200](https://github.com/jaegertracing/jaeger-client-java/pull/200) ([pavolloffay](https://github.com/pavolloffay))
- Fix conversion of ip address to int [\#199](https://github.com/jaegertracing/jaeger-client-java/pull/199) ([mabn](https://github.com/mabn))
- Add Process IP to jaeger.thrift [\#197](https://github.com/jaegertracing/jaeger-client-java/pull/197) ([black-adder](https://github.com/black-adder))
- Clean after crossdock [\#195](https://github.com/jaegertracing/jaeger-client-java/pull/195) ([pavolloffay](https://github.com/pavolloffay))
- HTTP sender for jaeger.thrift [\#194](https://github.com/jaegertracing/jaeger-client-java/pull/194) ([pavolloffay](https://github.com/pavolloffay))
- Changelog for 0.19.0 [\#191](https://github.com/jaegertracing/jaeger-client-java/pull/191) ([pavolloffay](https://github.com/pavolloffay))
- Save baggage in span [\#189](https://github.com/jaegertracing/jaeger-client-java/pull/189) ([black-adder](https://github.com/black-adder))
- Upgrade to OpenTracing java 0.30.0 with in-process propagation support [\#188](https://github.com/jaegertracing/jaeger-client-java/pull/188) ([objectiser](https://github.com/objectiser))
- Refactor RemoteReporter to speed up tests [\#187](https://github.com/jaegertracing/jaeger-client-java/pull/187) ([mabn](https://github.com/mabn))
- Fix Flaky RemoteReporter test [\#185](https://github.com/jaegertracing/jaeger-client-java/pull/185) ([mabn](https://github.com/mabn))
- Remove unnecessary gradle definition [\#183](https://github.com/jaegertracing/jaeger-client-java/pull/183) ([pavolloffay](https://github.com/pavolloffay))
- Normalize Configuration.fromEnv\(\) methods [\#181](https://github.com/jaegertracing/jaeger-client-java/pull/181) ([yurishkuro](https://github.com/yurishkuro))
- Handle full queue in RemoteReporter [\#180](https://github.com/jaegertracing/jaeger-client-java/pull/180) ([mabn](https://github.com/mabn))
- \[jaeger-core:test\] testRemoteReporterReport failing randomly [\#177](https://github.com/jaegertracing/jaeger-client-java/pull/177) ([jkandasa](https://github.com/jkandasa))
- Remove boilerplate from gradle scripts [\#176](https://github.com/jaegertracing/jaeger-client-java/pull/176) ([pavolloffay](https://github.com/pavolloffay))
- Initial implementation of a TracerResolver for Jaeger [\#175](https://github.com/jaegertracing/jaeger-client-java/pull/175) ([objectiser](https://github.com/objectiser))
- Do not truncate reported strings [\#174](https://github.com/jaegertracing/jaeger-client-java/pull/174) ([pavolloffay](https://github.com/pavolloffay))
- remove formatter comments [\#167](https://github.com/jaegertracing/jaeger-client-java/pull/167) ([pavolloffay](https://github.com/pavolloffay))
- Issue\#165 fix compile warnings [\#166](https://github.com/jaegertracing/jaeger-client-java/pull/166) ([kevinearls](https://github.com/kevinearls))
- Bump OpenTracing to 0.22.0 [\#164](https://github.com/jaegertracing/jaeger-client-java/pull/164) ([pavolloffay](https://github.com/pavolloffay))
- Fix endtoend crossdock tests [\#163](https://github.com/jaegertracing/jaeger-client-java/pull/163) ([black-adder](https://github.com/black-adder))
- Fix for Issue\#11 increase test coverage for jaeger-jaxrs2 tests [\#162](https://github.com/jaegertracing/jaeger-client-java/pull/162) ([kevinearls](https://github.com/kevinearls))
- Multiple references clarification [\#161](https://github.com/jaegertracing/jaeger-client-java/pull/161) ([pavolloffay](https://github.com/pavolloffay))
- Change jaeger UDP sender to use agent compact port [\#160](https://github.com/jaegertracing/jaeger-client-java/pull/160) ([pavolloffay](https://github.com/pavolloffay))
- Checkstyle fix for tests [\#158](https://github.com/jaegertracing/jaeger-client-java/pull/158) ([pavolloffay](https://github.com/pavolloffay))
- SpanContext add deprecated \*\*ID methods [\#157](https://github.com/jaegertracing/jaeger-client-java/pull/157) ([pavolloffay](https://github.com/pavolloffay))
- Fix immutable baggage when returning iterator from context [\#156](https://github.com/jaegertracing/jaeger-client-java/pull/156) ([pavolloffay](https://github.com/pavolloffay))
- Make build parallel [\#155](https://github.com/jaegertracing/jaeger-client-java/pull/155) ([vprithvi](https://github.com/vprithvi))
- Add endtoend crossdock [\#154](https://github.com/jaegertracing/jaeger-client-java/pull/154) ([black-adder](https://github.com/black-adder))
- Add codecov ignores [\#153](https://github.com/jaegertracing/jaeger-client-java/pull/153) ([vprithvi](https://github.com/vprithvi))
- Add license test to gradle checks [\#151](https://github.com/jaegertracing/jaeger-client-java/pull/151) ([vprithvi](https://github.com/vprithvi))
- Bump gradle version to 3.5 [\#150](https://github.com/jaegertracing/jaeger-client-java/pull/150) ([pavolloffay](https://github.com/pavolloffay))
- Google checkstyle with gradle checkstyle plugin [\#149](https://github.com/jaegertracing/jaeger-client-java/pull/149) ([pavolloffay](https://github.com/pavolloffay))
- Enable process level tags to be associated with the tracer [\#143](https://github.com/jaegertracing/jaeger-client-java/pull/143) ([objectiser](https://github.com/objectiser))
- Use jaeger.thrift [\#142](https://github.com/jaegertracing/jaeger-client-java/pull/142) ([pavolloffay](https://github.com/pavolloffay))
- Extract jaeger-b3 from jaeger-zipkin w/o any non-core deps [\#141](https://github.com/jaegertracing/jaeger-client-java/pull/141) ([adriancole](https://github.com/adriancole))
- Java6 fix [\#138](https://github.com/jaegertracing/jaeger-client-java/pull/138) ([felixbarny](https://github.com/felixbarny))
- Upgrade to opentracing-java 0.21.0 [\#137](https://github.com/jaegertracing/jaeger-client-java/pull/137) ([objectiser](https://github.com/objectiser))
- Add GuaranteedThroughputSampler\#update [\#134](https://github.com/jaegertracing/jaeger-client-java/pull/134) ([black-adder](https://github.com/black-adder))
- Java 6 compatibility [\#132](https://github.com/jaegertracing/jaeger-client-java/pull/132) ([felixbarny](https://github.com/felixbarny))
- Bump opentracingVersion to latest [\#131](https://github.com/jaegertracing/jaeger-client-java/pull/131) ([black-adder](https://github.com/black-adder))
- Add converter from internal span to Jaeger span [\#130](https://github.com/jaegertracing/jaeger-client-java/pull/130) ([black-adder](https://github.com/black-adder))
- Pin errorprone version and thus javac version [\#128](https://github.com/jaegertracing/jaeger-client-java/pull/128) ([vprithvi](https://github.com/vprithvi))
- Fix leaky bucket bug [\#127](https://github.com/jaegertracing/jaeger-client-java/pull/127) ([black-adder](https://github.com/black-adder))
- Clean up crossdock tests [\#126](https://github.com/jaegertracing/jaeger-client-java/pull/126) ([black-adder](https://github.com/black-adder))
- Fix existing license to 2016 [\#125](https://github.com/jaegertracing/jaeger-client-java/pull/125) ([black-adder](https://github.com/black-adder))
- Run RemoteReporter's flush timer in daemon thread. [\#124](https://github.com/jaegertracing/jaeger-client-java/pull/124) ([aristidisp](https://github.com/aristidisp))
- Support multiple tracers for adaptive sampling xdock tests [\#123](https://github.com/jaegertracing/jaeger-client-java/pull/123) ([black-adder](https://github.com/black-adder))
- Add xdock client service [\#122](https://github.com/jaegertracing/jaeger-client-java/pull/122) ([black-adder](https://github.com/black-adder))
- Remove sampling strategy [\#118](https://github.com/jaegertracing/jaeger-client-java/pull/118) ([vprithvi](https://github.com/vprithvi))
- Refactor tests for HttpSamplingManager [\#117](https://github.com/jaegertracing/jaeger-client-java/pull/117) ([vprithvi](https://github.com/vprithvi))
- Add tests for RemoteControlledSampler [\#115](https://github.com/jaegertracing/jaeger-client-java/pull/115) ([vprithvi](https://github.com/vprithvi))
- Add codecov-io for code coverage [\#113](https://github.com/jaegertracing/jaeger-client-java/pull/113) ([vprithvi](https://github.com/vprithvi))
- Support per operation sampling [\#112](https://github.com/jaegertracing/jaeger-client-java/pull/112) ([vprithvi](https://github.com/vprithvi))
- Refactor Sampler interface [\#111](https://github.com/jaegertracing/jaeger-client-java/pull/111) ([vprithvi](https://github.com/vprithvi))
- Use @Slf4j annotation wherever possible [\#105](https://github.com/jaegertracing/jaeger-client-java/pull/105) ([vprithvi](https://github.com/vprithvi))
- Update mockito to 2.2.28 [\#104](https://github.com/jaegertracing/jaeger-client-java/pull/104) ([vprithvi](https://github.com/vprithvi))
- Define json models for SamplingStrategyResponse and use them [\#103](https://github.com/jaegertracing/jaeger-client-java/pull/103) ([vprithvi](https://github.com/vprithvi))
- Fix client/server span classification [\#102](https://github.com/jaegertracing/jaeger-client-java/pull/102) ([vprithvi](https://github.com/vprithvi))
- Make Tracer\#isRPCServer robust [\#101](https://github.com/jaegertracing/jaeger-client-java/pull/101) ([vprithvi](https://github.com/vprithvi))
- Tighten Span\#isRPC check [\#100](https://github.com/jaegertracing/jaeger-client-java/pull/100) ([vprithvi](https://github.com/vprithvi))
- Add license header [\#99](https://github.com/jaegertracing/jaeger-client-java/pull/99) ([vprithvi](https://github.com/vprithvi))
- Automatically add span names for jersey instrumentation [\#95](https://github.com/jaegertracing/jaeger-client-java/pull/95) ([vprithvi](https://github.com/vprithvi))
- Bumps to zipkin 1.16.2 [\#93](https://github.com/jaegertracing/jaeger-client-java/pull/93) ([adriancole](https://github.com/adriancole))
- Fix code coverage generation [\#92](https://github.com/jaegertracing/jaeger-client-java/pull/92) ([vprithvi](https://github.com/vprithvi))
- Add Builder to TextMapCodec \(https://github.com/uber/jaeger-client-java/issues/88\) [\#91](https://github.com/jaegertracing/jaeger-client-java/pull/91) ([hzariv](https://github.com/hzariv))
- Add Configuration.closeTracer [\#90](https://github.com/jaegertracing/jaeger-client-java/pull/90) ([vprithvi](https://github.com/vprithvi))
- Publish a shadow jar with thrift 0.9.2 [\#83](https://github.com/jaegertracing/jaeger-client-java/pull/83) ([vprithvi](https://github.com/vprithvi))
- Add init logging to tracer [\#82](https://github.com/jaegertracing/jaeger-client-java/pull/82) ([vprithvi](https://github.com/vprithvi))
- Add a Tracer.close\(\) method [\#78](https://github.com/jaegertracing/jaeger-client-java/pull/78) ([vprithvi](https://github.com/vprithvi))
- Fix .gitignore [\#77](https://github.com/jaegertracing/jaeger-client-java/pull/77) ([vprithvi](https://github.com/vprithvi))
- Bumps zipkin dependency versions [\#76](https://github.com/jaegertracing/jaeger-client-java/pull/76) ([adriancole](https://github.com/adriancole))
- set host on all zipkin annotations for queries [\#75](https://github.com/jaegertracing/jaeger-client-java/pull/75) ([rubbish](https://github.com/rubbish))
- allow runnables without current span to run [\#74](https://github.com/jaegertracing/jaeger-client-java/pull/74) ([rubbish](https://github.com/rubbish))
- Downgrade to thrift 0.9.2 [\#73](https://github.com/jaegertracing/jaeger-client-java/pull/73) ([vprithvi](https://github.com/vprithvi))
- Fix http url tag on jaxrs2 filters and add tests [\#72](https://github.com/jaegertracing/jaeger-client-java/pull/72) ([vprithvi](https://github.com/vprithvi))
- Fix typo in configuration and add convenience setter for metricRegistry [\#71](https://github.com/jaegertracing/jaeger-client-java/pull/71) ([vprithvi](https://github.com/vprithvi))
- Clean-up README and javadoc [\#69](https://github.com/jaegertracing/jaeger-client-java/pull/69) ([yurishkuro](https://github.com/yurishkuro))
- Add unit test to check that RPC spans share the same span ID [\#68](https://github.com/jaegertracing/jaeger-client-java/pull/68) ([yurishkuro](https://github.com/yurishkuro))
- Remove call to Github API to get current branch name [\#67](https://github.com/jaegertracing/jaeger-client-java/pull/67) ([breerly](https://github.com/breerly))
- Fix bug in dropwizard configuration [\#66](https://github.com/jaegertracing/jaeger-client-java/pull/66) ([vprithvi](https://github.com/vprithvi))
- Expose interceptor callbacks and custom operation name for extension classes [\#65](https://github.com/jaegertracing/jaeger-client-java/pull/65) ([atanasbalevski](https://github.com/atanasbalevski))
- Refactor and update readme [\#64](https://github.com/jaegertracing/jaeger-client-java/pull/64) ([vprithvi](https://github.com/vprithvi))
- Add support for Apache Http Client [\#63](https://github.com/jaegertracing/jaeger-client-java/pull/63) ([vprithvi](https://github.com/vprithvi))
- Fix crossdock not implemetned error propagation [\#60](https://github.com/jaegertracing/jaeger-client-java/pull/60) ([oibe](https://github.com/oibe))
- Add errorprone plugin [\#59](https://github.com/jaegertracing/jaeger-client-java/pull/59) ([vprithvi](https://github.com/vprithvi))
- Refactor configuration [\#57](https://github.com/jaegertracing/jaeger-client-java/pull/57) ([vprithvi](https://github.com/vprithvi))
- Use the verify task in google-formatter instead of checkstyle [\#55](https://github.com/jaegertracing/jaeger-client-java/pull/55) ([vprithvi](https://github.com/vprithvi))
- Add google-formatter [\#54](https://github.com/jaegertracing/jaeger-client-java/pull/54) ([vprithvi](https://github.com/vprithvi))
- Support debug traces via HTTP header "jaeger-debug-id" [\#53](https://github.com/jaegertracing/jaeger-client-java/pull/53) ([yurishkuro](https://github.com/yurishkuro))
- Tolerate reads of 128 bit X-B3-TraceId [\#52](https://github.com/jaegertracing/jaeger-client-java/pull/52) ([adriancole](https://github.com/adriancole))
- Fix checkstyle and use google style guide [\#51](https://github.com/jaegertracing/jaeger-client-java/pull/51) ([vprithvi](https://github.com/vprithvi))
- Introduce Clock interface for time measurements [\#50](https://github.com/jaegertracing/jaeger-client-java/pull/50) ([yurishkuro](https://github.com/yurishkuro))
- Exclude test classes from published jars [\#49](https://github.com/jaegertracing/jaeger-client-java/pull/49) ([yurishkuro](https://github.com/yurishkuro))
- Default peer service name to blank as it is required in Thrift Endpoint [\#47](https://github.com/jaegertracing/jaeger-client-java/pull/47) ([yurishkuro](https://github.com/yurishkuro))
- Protect against empty service name [\#46](https://github.com/jaegertracing/jaeger-client-java/pull/46) ([yurishkuro](https://github.com/yurishkuro))
- Fix typo in rules.mk [\#43](https://github.com/jaegertracing/jaeger-client-java/pull/43) ([vprithvi](https://github.com/vprithvi))
- Record sampling strategy as span tags [\#40](https://github.com/jaegertracing/jaeger-client-java/pull/40) ([yurishkuro](https://github.com/yurishkuro))
- Updates to zipkin reporter 0.4 [\#39](https://github.com/jaegertracing/jaeger-client-java/pull/39) ([adriancole](https://github.com/adriancole))
- Fix compileThrift in jaeger-core [\#37](https://github.com/jaegertracing/jaeger-client-java/pull/37) ([vprithvi](https://github.com/vprithvi))
- add python image to crossdock test and update go image to use master … [\#36](https://github.com/jaegertracing/jaeger-client-java/pull/36) ([black-adder](https://github.com/black-adder))
- Adds jaeger-zipkin which makes Jaeger compatible with Zipkin tracing [\#34](https://github.com/jaegertracing/jaeger-client-java/pull/34) ([adriancole](https://github.com/adriancole))
- Implement Crossdock TChannel requests [\#33](https://github.com/jaegertracing/jaeger-client-java/pull/33) ([yurishkuro](https://github.com/yurishkuro))
- Rename jaeger-java-crossdock to jaeger-crossdock [\#32](https://github.com/jaegertracing/jaeger-client-java/pull/32) ([yurishkuro](https://github.com/yurishkuro))
- Use internal span in Reporters [\#31](https://github.com/jaegertracing/jaeger-client-java/pull/31) ([yurishkuro](https://github.com/yurishkuro))
- Run thrift compiler from docker image [\#28](https://github.com/jaegertracing/jaeger-client-java/pull/28) ([yurishkuro](https://github.com/yurishkuro))
- Upgrade to latest OpenTracing API \(v.0.13\) [\#27](https://github.com/jaegertracing/jaeger-client-java/pull/27) ([yurishkuro](https://github.com/yurishkuro))
- Consolidate dependency versions management [\#26](https://github.com/jaegertracing/jaeger-client-java/pull/26) ([yurishkuro](https://github.com/yurishkuro))
- update Jaeger readme with instrumentation information [\#23](https://github.com/jaegertracing/jaeger-client-java/pull/23) ([oibe](https://github.com/oibe))
- fix travis [\#20](https://github.com/jaegertracing/jaeger-client-java/pull/20) ([oibe](https://github.com/oibe))
- add logging [\#19](https://github.com/jaegertracing/jaeger-client-java/pull/19) ([oibe](https://github.com/oibe))
- equals comparing reference \#14 [\#18](https://github.com/jaegertracing/jaeger-client-java/pull/18) ([nklmish](https://github.com/nklmish))
- add jaeger-crossdock to opensource [\#17](https://github.com/jaegertracing/jaeger-client-java/pull/17) ([oibe](https://github.com/oibe))
- Add dropwizard [\#16](https://github.com/jaegertracing/jaeger-client-java/pull/16) ([oibe](https://github.com/oibe))
- updated .gitignore to cover: [\#13](https://github.com/jaegertracing/jaeger-client-java/pull/13) ([nklmish](https://github.com/nklmish))
- Add jaxrs2, and port internal changes [\#8](https://github.com/jaegertracing/jaeger-client-java/pull/8) ([oibe](https://github.com/oibe))
- add javadoc, and sources  artifacts for sonatype publishing [\#6](https://github.com/jaegertracing/jaeger-client-java/pull/6) ([oibe](https://github.com/oibe))
- Open source jaeger-core [\#1](https://github.com/jaegertracing/jaeger-client-java/pull/1) ([yurishkuro](https://github.com/yurishkuro))

\* *This Change Log was automatically generated by [github_changelog_generator](https://github.com/skywinder/Github-Changelog-Generator)*

**Comment by pavolloffay** [comment]
@jaegertracing/jaeger-maintainers what do you think about this using in repositories where we don't list breaking changes?

**Comment by objectiser** [comment]
Maybe we can have a convention of including `[BREAKING CHANGE]` as a prefix to PR titles?

**Comment by pavolloffay** [comment]
That is the same as putting them in a separate section. I would prefer a separate section.

My concern is that we cannot add migration steps directly do the changelog as it will be overridden.
